insights/core/__init__.py

```diff
@@ -1542,3 +1542,150 @@ def __init__(self, *args, **kwargs):
+        depends:        ptp
+        intree:         Y
+        vermagic:       3.10.0-993.el7.x86_64 SMP mod_unload modversions
+        signer:         Red Hat Enterprise Linux kernel signing key
+        sig_key:        81:7C:CB:07:72:4E:7F:B8:15:24:10:F9:27:2D:AA:CF:80:3E:CE:59
+        sig_hashalgo:   sha256
+        parm:           debug:Debug level (0=none,...,16=all), Debug mask (0x8XXXXXXX) (uint)
+
+    Examples:
+        >>> modinfo_obj
+        <insights.parsers.modinfo_i40e.ModInfo_i40e object at 0x7f722d92ff28>
+        >>> modinfo_obj.module_name
+        'i40e'
+        >>> modinfo_obj.module_version
+        '2.3.2-k'
+        >>> modinfo_obj.module_path
+        "/lib/modules/3.10.0-993.el7.x86_64/kernel/drivers/net/ethernet/intel/i40e/i40e.ko.xz"
+    """
+    def parse_content(self, content):
+        self.data = {}
+        self._module_deps = ''
+        self._module_name = ''
+        self._module_path = ''
+        self._firmware_path = ''
+        alias_key = ''
+        fw_key = ''
+        parm_key = ''
+
+        for line in content:
+            line = line.strip()
+            key = line.split(':')[0]
+            value = line.split()[1]
+            if key and value and key == 'filename':
+                self._module_path = value
+                self._module_name = value.rsplit("/")[-1].split('.')[0]
+            elif key and value and key == 'sig_key':
+                self.data[key] = value
+            elif key and value and key == 'depends':
+                self._module_deps = [mod for mod in value.split(',')]
+                self.data[key] = self._module_deps
+            elif key and value and key == 'parm' and parm_key == key:
+                value = ' '.join(line.split()[1:])
+                self.data[key].append(value)
+            elif key and value and key == 'alias' and alias_key == key:
+                self.data[key].append(value)
+            elif key and value and key == 'firmware' and fw_key == key:
+                self.data[key].append(value)
+            elif key and value and key == 'alias':
+                self.data[key] = []
+                self.data[key].append(value)
+                alias_key = key
+            elif key and value and key == 'firmware':
+                self.data[key] = []
+                value = ' '.join(line.split()[1:])
+                self.data[key].append(value)
+                fw_key = key
+            elif key and value and key == 'parm':
+                self.data[key] = []
+                value = ' '.join(line.split()[1:])
+                self.data[key].append(value)
+                parm_key = key
+            else:
+                value = ' '.join(line.split()[1:])
+                self.data[key] = value
+
+    def __contains__(self, option):
+        """
+        (bool): This will return True if `option` is present kernel info when set, else False
+        """
+        return option in self.data.keys() if self.data else False
```

Comment:
1. A `SkipException` or `ParseException` should be raised when `self.data` is empty
2. You do not need to get the `.keys()` of the dict, `option in self.data` is the equivalent way.

Reply:
Done